src/elm/Data/DataSource.elm

```diff
@@ -0,0 +1,102 @@
+module Data.DataSource
+    exposing
+        ( DataSourceId(..)
+        , DataSource
+        , decode
+        , decodeList
+        , encodeId
+        )
+
+import Json.Decode as Decode exposing (Decoder)
+import Json.Encode
+
+
+type DataSourceId
+    = DataSourceId Int
+
+
+type Description
+    = NoDescription
+    | Description String
```

Comment:
Is there an added value to creating a custom type here rather than using `Maybe` ? It looks like you're using this type to denote an optional value?

Would:

```elm
type Description = Description String

type alias DataSource =
  { id : DataSourceId
  , title : String
  , description : Maybe Description
  }
```

solve the problem?